backend/v2/project_rewards/services.py

```diff
@@ -39,3 +54,115 @@ async def get(self) -> CappedQuadriaticFunding:
             allocations=allocations,
             matched_rewards=matched_rewards,
         )
+
+
+async def get_rewards_merkle_tree_for_epoch(
+    session: AsyncSession,
+    epoch_number: int,
+) -> RewardsMerkleTreeResponseV1 | None:
+    """
+    Get the rewards merkle tree for a given epoch.
+    """
+
+    # Merkle tree is based on rewards, so we need to get them first
+    rewards = await get_rewards_for_epoch(session, epoch_number)
+    if not rewards:
+        return None
+
+    # Build the merkle tree (using the leaf encoding)
+    LEAF_ENCODING: list[str] = ["address", "uint256"]
+    mt = StandardMerkleTree.of(
+        [[leaf.address, int(leaf.amount)] for leaf in rewards],
+        LEAF_ENCODING,
+    )
+
+    # Build response model
+    leaves = [
+        RewardsMerkleTreeLeafV1(address=leaf.value[0], amount=leaf.value[1])
+        for leaf in mt.values
+    ]
+
+    rewards_sum = sum(leaf.amount for leaf in leaves)
+
+    return RewardsMerkleTreeResponseV1(
+        epoch=epoch_number,
+        rewards_sum=rewards_sum,
+        root=mt.root,
+        leaves=leaves,
+        leaf_encoding=LEAF_ENCODING,
+    )
+
+
+def calculate_octant_rewards(
+    eth_proceeds: int,
+    total_effective_deposit: int,
+) -> OctantRewardsDTO:
+    """
+    Calculate the octant rewards based on ETH proceeds and total effective deposit.
+    Proceeds are the what the octant gets from the staking.
+    Effective deposit is the amount of GLMs that are eligible for rewards.
+
+    Parameters:
+        eth_proceeds: int - The amount of ETH proceeds.
+        total_effective_deposit: int - The total amount of GLMs deposited.
+
+    Returns:
+        OctantRewardsDTO - The octant rewards distribution.
+    """
+
+    settings = OctantRewardsSettings()
+    rewards = calculate_rewards(settings, eth_proceeds, total_effective_deposit)
+
+    # fmt: off
+    return OctantRewardsDTO(
```

Comment:
fancy but is it worth to turn off the formatter to have these values only justified in the code? 🚀